### PR TITLE
Update installer 

### PR DESCRIPTION
**Pull request by thudoan1706**

### CODE REVIEW

**Review by hykilpikonna — APPROVED** [review]

File: src/main/java/org/hydev/mcpm/client/DatabaseManager.java [hunk 1]
```diff
@@ -1,35 +1,37 @@
...
 import org.hydev.mcpm.client.installer.InstallResult;
 import org.hydev.mcpm.client.installer.input.InstallInput;
 import org.hydev.mcpm.client.models.PluginVersion;
 import org.hydev.mcpm.client.models.PluginYml;
 
+import java.net.URI;
 import java.util.List;
 
 /**
  * Database API
```
Comment (MstrPikachu) [inline src/main/java/org/hydev/mcpm/client/DatabaseManager.java:19]: if you could mention a bit more about what the Database API does that would be great

File: src/main/java/org/hydev/mcpm/client/DatabaseManager.java [hunk 2]
```diff
@@ -39,16 +41,35 @@ public SearchPackagesResult getSearchResult(InstallInput input) {
...
+
+
+    /**
+     * check if plugin with given name already installed locally
      *
      * @param pluginVersion The version of the installed plugin
      * */
-    public boolean checkPluginInstalled(PluginVersion pluginVersion) {
+    public boolean checkPluginInstalledByVersion(PluginVersion pluginVersion) {
         var name = pluginVersion.meta().name();
```
Comment (MstrPikachu) [inline src/main/java/org/hydev/mcpm/client/DatabaseManager.java:67]: name isn't used anywhere, did you forget to check the names are equal as well?

File: src/main/java/org/hydev/mcpm/client/arguments/CommandsFactory.java [hunk 3]
```diff
@@ -44,9 +46,16 @@ public static List<CommandParser> baseParsers() {
         var searchController = new SearchPackagesController(searcher);
         var mirrorController = new MirrorController(new MirrorSelector());
         var infoController = new InfoController(tracker);
+        PluginLoader pluginLoader = null;
+        if (isMinecraft) {
+            pluginLoader = new PluginLoader();
+        }
+        DatabaseManager databaseManager = new DatabaseManager(tracker, searcher);
+        System.out.println(isMinecraft);
```
Comment (MstrPikachu) [inline src/main/java/org/hydev/mcpm/client/arguments/CommandsFactory.java:54]: do we need to print here

File: src/main/java/org/hydev/mcpm/client/arguments/CommandsFactory.java [hunk 4]
```diff
@@ -44,9 +46,16 @@ public static List<CommandParser> baseParsers() {
...
         var installController = new InstallController(new InstallInteractor(
             new SpigotPluginDownloader(new Downloader(), host.toString()),
-            new DatabaseManager(tracker, searcher)));
+            databaseManager, pluginLoader));
```
Comment (MstrPikachu) [inline src/main/java/org/hydev/mcpm/client/arguments/CommandsFactory.java:57]: Can SpigotPluginDownloader handle a null pluginLoader or do we only care if it works in minecraft

File: src/main/java/org/hydev/mcpm/client/installer/InstallInteractor.java [hunk 5]
```diff
@@ -84,7 +96,19 @@ public InstallResult installPlugin(InstallInput installInput)
                 installPlugin(dependencyInput);
```
Comment (MstrPikachu) [inline src/main/java/org/hydev/mcpm/client/installer/InstallInteractor.java:96]: The recursive step doesn't use any results from the recursion. For example, if a dependency fails to be installed, the overall result would still return success if the initial plugin is a success. I can approve if we make this an issue and fix it later

File: src/main/java/org/hydev/mcpm/client/arguments/CommandsFactory.java (hunk 4 above)
Comment (hykilpikonna) [reply src/main/java/org/hydev/mcpm/client/arguments/CommandsFactory.java:57]: it should handle a null plugin loader 🤔

Comment (hykilpikonna) [reply src/main/java/org/hydev/mcpm/client/arguments/CommandsFactory.java:57]: Or maybe we can create a plugin loader implementation that does nothing?

**Review by MstrPikachu — APPROVED** [review]
I'll make an issue

**Pull request merged**